pnl-subgraph/src/ConditionalTokensMapping.ts

```diff
@@ -0,0 +1,181 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+
+import { BigInt, log } from '@graphprotocol/graph-ts';
+
+import {
+  ConditionPreparation,
+  ConditionResolution,
+  PositionSplit,
+  PositionsMerge,
+  PayoutRedemption,
+} from './types/ConditionalTokens/ConditionalTokens';
+
+import { updateUserPositionWithSell } from './utils/updateUserPositionWithSell';
+import { updateUserPositionWithBuy } from './utils/updateUserPositionWithBuy';
+import { loadOrCreateUserPosition } from './utils/loadOrCreateUserPosition';
+
+import { computePositionId } from '../../common';
+import {
+  COLLATERAL_SCALE,
+  EXCHANGE,
+  NEG_RISK_ADAPTER,
+  USDC,
+} from '../../common/constants';
+import { createCondition } from './utils/createCondition';
+import { loadCondition } from './utils/loadCondition';
+
+// SPLIT
+export function handlePositionSplit(event: PositionSplit): void {
+  const conditionId = event.params.conditionId;
+  const condition = loadCondition(conditionId);
+  if (condition == null) {
+    // ignore
+    return;
+  }
+
+  // - don't track splits from the NegRiskAdapter
+  //   these are handled in the NegRiskAdapterMapping
+  // - don't track splits from the CTFExchange
+  if (
+    [NEG_RISK_ADAPTER.toHexString(), EXCHANGE.toHexString()].includes(
+      event.params.stakeholder.toHexString(),
+    )
+  ) {
+    return;
+  }
+
+  const SELL_PRICE = COLLATERAL_SCALE.div(BigInt.fromI32(2));
+  // @ts-ignore
+  let outcomeIndex: u8 = 0;
+  for (; outcomeIndex < 2; outcomeIndex++) {
+    const positionId = computePositionId(
+      USDC,
+      event.params.conditionId,
+      outcomeIndex,
+    );
+
+    updateUserPositionWithSell(
+      event.params.stakeholder,
+      positionId,
+      SELL_PRICE,
+      event.params.amount,
+    );
+  }
+}
+
+// MERGE
+export function handlePositionsMerge(event: PositionsMerge): void {
+  const conditionId = event.params.conditionId;
+  const condition = loadCondition(conditionId);
+  if (condition == null) {
+    // ignore
+    return;
+  }
+
+  // - don't track merges from the NegRiskAdapter
+  //   these are handled in the NegRiskAdapterMapping
+  // - don't track merges from the CTFExchange
+  if (
+    [NEG_RISK_ADAPTER.toHexString(), EXCHANGE.toHexString()].includes(
+      event.params.stakeholder.toHexString(),
+    )
+  ) {
+    return;
+  }
+
+  const BUY_PRICE = COLLATERAL_SCALE.div(BigInt.fromI32(2));
+  // @ts-ignore
+  let outcomeIndex: u8 = 0;
+  for (; outcomeIndex < 2; outcomeIndex++) {
+    const positionId = computePositionId(
+      USDC,
+      event.params.conditionId,
+      outcomeIndex,
+    );
+
+    updateUserPositionWithBuy(
```

Comment:
I would think merge = sell (output cash).

Reply:
you are correct sir